02_Development/Display/sensors.py

```diff
@@ -0,0 +1,157 @@
+        """Actual implementation"""
+
+        def __init__(self,
+                     pressure_sampling_rate,
+                     pressure_oversampling,
+                     temperature_sampling_rate,
+                     temperature_oversampling):
+            pass
+
+        def connected_sensors(self):
+            pass
+
+        def calibration_pressure_sensor_connected(self):
+            pass
+
+        def poll(self):
+            pass
+
+
+else:
+
+    class SensorsData(SensorsDataABC):
+        """Dummy class while I work on the real thing"""
+
+        def __init__(self,
+                     pressure_sampling_rate,
+                     pressure_oversampling,
+                     temperature_sampling_rate,
+                     temperature_oversampling):
+            self._fake_data = \
+                ProcessSampleData("TestData/20200609T2358Z_patrickData.txt")
+            self._data_index = 0
+
+        def connected_sensors(self, not_enough_sensors=False):
+            try:
+                if (os.environ[constants.SENSOR_QUANTITY]
+                        == constants.NOT_ENOUGH_SENSORS):
+                    sensors = tuple([(constants.PRESSURE_SENSOR,)]
+                                    + [(constants.PRESSURE_SENSOR,
+                                        constants.SENSIRION_SENSOR)
+                                       for _ in range(constants.
+                                                      NUMBER_OF_PATIENTS-1)
+                                       ])
+                    raise NotEnoughSensors(
+                        f"{len(self._tubes_with_enough_sensors(sensors))} "
+                        "tube(s) do not have both a pressure sensor "
+                        "and a flow sensor")
+
+                elif (os.environ[constants.SENSOR_QUANTITY]
+                      == constants.TOO_MANY_SENSORS):
+                    return tuple([(constants.PRESSURE_SENSOR,
+                                   constants.SENSIRION_SENSOR,
+                                   constants.MASS_AIRFLOW_SENSOR)]
+                                 + [(constants.PRESSURE_SENSOR,
+                                     constants.SENSIRION_SENSOR)
+                                    for _ in range(constants.
+                                                   NUMBER_OF_PATIENTS-1)
+                                    ])
+            except KeyError:
+                pass
+            return tuple((constants.PRESSURE_SENSOR,
+                          constants.SENSIRION_SENSOR)
+                         for _ in range(constants.NUMBER_OF_PATIENTS))
+
+        def calibration_pressure_sensor_connected(self, fail=False):
+            if fail:
+                return False
+            else:
+                return True
+
+        def poll(self):
+            """Pulls data from the pressure and flow sensors"""
+            self._data_index += 1
+            return tuple((self._fake_data.pressures[self._data_index-1],
```

Comment:
why the -1 here?  Can this give you a -1 index?

Reply:
`self._data_index` is initially 0, so this will never give -1.  The reason I've done it this way is to make it so it iterates through the data files invisibly.  But the way I've written it is unclear so I'll revise it.